### ISSUE TITLE
[Bug Report] Gallery Order gets messed up in wall view and lightbox

### ISSUE THREAD

**Issue by yoshnopa** [stashapp/stash#2376]
**Describe the bug**
In the most current version v0.13.0-2-ga0595282 the gallery wall view is not in the order that one would expect. I didn't even find a pattern what exactly happens here. For quick info, I am only using .cbz files for galleries, so maybe this is for those files only. 
This messed up order (and cover image as well) is also continued when clicking on a wall item to show the light box. Interestingly enough, when going into the specific gallery (the view you get into by clicking in the grid view) the order is still respected and also the lightbox started here with the wall view respects the galleries order.
The front page (cover) is also wrong in the grid and list view, and consistently so (between views or sorting orders).
About my .cbz files, they contain their images numbered pretty consistently, so basically from 001.jpg to 999.jpg for example. I have never found a program that wasn't able to figure out the order of that.
I was able to pin down that it was down after version v0.12.0-82-g9ef30604, so in terms of commits it has to come from one of those:
https://github.com/stashapp/stash/pull/2353
https://github.com/stashapp/stash/pull/2368
https://github.com/stashapp/stash/pull/2372
https://github.com/stashapp/stash/pull/2363

**To Reproduce**
1. Go into Galleries
2. Choose the Wall option
3. Open lightbox
4. Open other view-options in Galleries

**Expected behavior**
Front cover being the first image in the .cbz, further images in the lightbox being in the alphanumerical order of the file names if not defined otherwise.

**Stash Version: (from Settings -> About):**
v0.13.0-2-ga0595282

**Desktop (please complete the following information):**
 - OS: Ubuntu 21.10
 - Browser Chromium
 - Version 98.0.4758.102

**Comment by yoshnopa** [comment]
Thank you for your quick bugfix!
Unfortunately, this does not resolve the issue in its entirety: Right now, the lightbox order is fixed and correct, but the viewed front cover is still not correct:
This means when browsing galleries, the viewed image is some seemingly random image from the gallery (seemingly because it is consistent between reloads, sortings and the like). When clicking on an image in the wall view, the first picture of the gallery is displayed instead of the one that was shown in the wall view (so this was corrected from when I opened the issue). 
This Bug is consistent between different views (Grid, Wall, List) and sortings (random, updated at etc), so its probably some wrong retrieval of data.
